### ISSUE TITLE
NoSuchMethodException: ... JavaModel.flushExternalFileCache(java.lang.Void) when building project using Java2Script builder

### ISSUE THREAD

**Issue by abego** [zhourenjian/java2script#3]
Java2Script throw an `NoSuchMethodException` for the method `org.eclipse.jdt.internal.core.JavaModel.flushExternalFileCache(java.lang.Void)`

when building a project using the Java2Script Builder.

# Steps to reproduce

For a project using the Java2Script Builder create an Ant file and use the internal Ant task “[eclipse.incrementalBuild](http://help.eclipse.org/juno/index.jsp?topic=/org.eclipse.platform.doc.isv/guide/ant_eclipse_tasks.htm)” to trigger an Eclipse build. E.g.
```
	...
	<target name="build">
		<eclipse.incrementalBuild project="MyProject" kind="full"/>
	</target>
	...
```

Running the build target will then display a stack trace in the console, like this:

	[eclipse.incrementalBuild] java.lang.NoSuchMethodException: org.eclipse.jdt.internal.core.JavaModel.flushExternalFileCache(java.lang.Void)
	[eclipse.incrementalBuild] 	at java.lang.Class.getMethod(Class.java:1786)
	[eclipse.incrementalBuild] 	at net.sf.j2s.core.builder.JavaBuilder.initializeBuilder(JavaBuilder.java:618)
	[eclipse.incrementalBuild] 	at net.sf.j2s.core.builder.JavaBuilder.build(JavaBuilder.java:168)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:734)
	[eclipse.incrementalBuild] 	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:205)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:245)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:300)
	[eclipse.incrementalBuild] 	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:303)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:262)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:402)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.resources.Project$1.run(Project.java:556)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2241)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.resources.Project.internalBuild(Project.java:534)
	[eclipse.incrementalBuild] 	at org.eclipse.core.internal.resources.Project.build(Project.java:111)
	[eclipse.incrementalBuild] 	at org.eclipse.core.resources.ant.IncrementalBuild.execute(IncrementalBuild.java:80)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	[eclipse.incrementalBuild] 	at sun.reflect.GeneratedMethodAccessor62.invoke(Unknown Source)
	[eclipse.incrementalBuild] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	[eclipse.incrementalBuild] 	at java.lang.reflect.Method.invoke(Method.java:497)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.Task.perform(Task.java:348)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.Target.execute(Target.java:435)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	[eclipse.incrementalBuild] 	at org.eclipse.ant.internal.core.ant.EclipseDefaultExecutor.executeTargets(EclipseDefaultExecutor.java:36)
	[eclipse.incrementalBuild] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	[eclipse.incrementalBuild] 	at org.eclipse.ant.internal.core.ant.InternalAntRunner.run(InternalAntRunner.java:705)
	[eclipse.incrementalBuild] 	at org.eclipse.ant.internal.core.ant.InternalAntRunner.run(InternalAntRunner.java:527)
	[eclipse.incrementalBuild] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	[eclipse.incrementalBuild] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	[eclipse.incrementalBuild] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	[eclipse.incrementalBuild] 	at java.lang.reflect.Method.invoke(Method.java:497)
	[eclipse.incrementalBuild] 	at org.eclipse.ant.core.AntRunner.run(AntRunner.java:371)
	[eclipse.incrementalBuild] 	at org.eclipse.ant.internal.launching.launchConfigurations.AntLaunchDelegate$1.run(AntLaunchDelegate.java:269)
	[eclipse.incrementalBuild] 	at java.lang.Thread.run(Thread.java:745)

**Comment by abego** [comment]
Provided fix with pull request https://github.com/zhourenjian/java2script/pull/4

**Comment by zhourenjian** [comment]
Fixed in 4.5 and 4.6
